utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -311,13 +314,56 @@
                    The colon separating the "of" label and the original
                    declaration name.
                    '''),
-             Child('Original', kind='FunctionDeclName',
-                   description='The referenced original declaration.'),
+             Child('OriginalDeclName', kind='QualifiedDeclName',
+                   description='The referenced original declaration name.'),
              Child('Comma', kind='CommaToken', is_optional=True),
              Child('DiffParams', kind='DifferentiationParamsClause',
                    is_optional=True),
          ]),
 
+    # An optionally qualified declaration name.
+    # Currently used only for `@derivative` and `@transpose` attribute.
+    # TODO(TF-1066): Use module qualified name syntax/parsing instead of custom
+    # qualified name syntax/parsing.
+    #
+    # qualified-decl-name ->
+    #     base-type? '.'? (identifier | operator) decl-name-arguments?
+    # base-type ->
+    #     member-type-identifier | base-type-identifier
+    Node('QualifiedDeclName', kind='Syntax',
+         description='''
+         An optionally qualified function declaration name (e.g. `+(_:_:)`,
+         `A.B.C.foo(_:_:)`).
+         ''',
+         children=[
+             Child('BaseType', kind='Type', description='''
+                   The base type of the qualified name, optionally specified.
+                   ''',
+                   node_choices=[
+                       Child('MemberType', kind='MemberTypeIdentifier'),
+                       Child('SimpleType', kind='SimpleTypeIdentifier'),
+                   ], is_optional=True),
+             Child('Dot', kind='Token',
+                   token_choices=[
+                       'PeriodToken', 'PrefixPeriodToken'
+                   ], is_optional=True),
+             Child('Name', kind='Syntax', description='''
+                   The base name of the referenced function.
+                   ''',
+                   node_choices=[
+                       Child('Identifier', kind='IdentifierToken'),
+                       Child('PrefixOperator', kind='PrefixOperatorToken'),
+                       Child('PostfixOperator', kind='PostfixOperatorToken'),
+                       Child('SpacedBinaryOperator',
+                             kind='SpacedBinaryOperatorToken'),
+                   ]),
```

Comment:
```python
    kind='Token',
    token_choices=[
        'IdentifierToken',
        'UnspacedBinaryOperatorToken',
        'SpacedBinaryOperatorToken',
        'PrefixOperatorToken',
        'PostfixOperatorToken',
    ]),
```
This should work. e.g. `FunctionDeclSyntax.Identifier`

Reply:
Done in https://github.com/apple/swift/pull/27545/commits/c736938911f0d2c4b5fd08968c365bc58e1c0539, thank you!

Reply:
Would you like to be added as a reviewer on future parsing/syntax PRs? I'm happy to do so!

Reply:
> Would you like to be added as a reviewer on future parsing/syntax PRs?

Yes. I appreciate it!

Reply:
> Would you like to be added as a reviewer on future parsing/syntax PRs?

Yes. I appreciate it

Reply:
Great, will do 🙂